### DISCUSSION TITLE
Deploying local assets.json possible?

### DISCUSSION THREAD

**Discussion by Jee-Hex** [uBlockOrigin/uBlock-issues#2846] [Q&A]
Okay so I have been trying to deploy a customized `assets.json` to my Firefox Portable profile using `policies.json` for a while now but without much success.

What I have tried:

1.

```json
{
  "policies": {
	"3rdparty": {
		"Extensions": {
			"uBlock0@raymondhill.net": {
				"adminSettings": {
					"assetsBootstrapLocation": "file:///Z:/path/to/assets.json",
					"userSettings": {
						"colorBlindFriendly": true
						}
					}
				}
			}
		}
	}
}
```

which doesn't work and I see the following in the browser console.

```
Security Error: Content at moz-extension://[uBO-webextension-uuid]/background.html may not load or link to file:///Z:/path/to/assets.json?_=2.
```

2.
```json
{
  "policies": {
	"3rdparty": {
		"Extensions": {
			"uBlock0@raymondhill.net": {
				"adminSettings": {
					"assetsBootstrapLocation": "data:application/json,{%content_of_assets_json_in_a_single_line%}",
					"userSettings": {
						"colorBlindFriendly": true
						}
					}
				}
			}
		}
	}
}
```

which sort of worked (when the `"`s were properly escaped), but only in the sense that uBO no longer uses the `assets.json` bundled within the xpi. Instead the contents of the assets.json were completely empty.

Does anyone know if it is even _theoretically_ possible to deploy a local `assets.json`? Obviously I could always just directly modify the .xpi file, but that would have meant uninstalling and reinstalling the extension every time the cdnURLs of some malware filters change.

**Comment by ghost** [comment]
Try `localhost`?
